sphinxcontrib/needs/warnings.py

```diff
@@ -50,10 +52,18 @@ def process_warnings(app, exception):
                 logger.info("  {}: passed".format(warning_name))
             else:
                 need_ids = [x["id"] for x in result]
-                logger.info("  {}: failed".format(warning_name))
-                logger.info("  \t\tfailed needs: {} ({})".format(len(need_ids), ", ".join(need_ids)))
-                logger.info("  \t\tused filter: {}".format(warning_filter))
-                warning_raised = True
+                if warnings_always_warn:
+                    for need_id in need_ids:
```

Comment:
This would print the errors for EACH need.
But we only need it once per warning.

Please use also the same code / code-format as in the else-statement, so that it is easier to maintain.